pkg/resources/grant_role.go

```diff
@@ -0,0 +1,175 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var grantRoleSchema = map[string]*schema.Schema{
+	"role_name": {
+		Type:             schema.TypeString,
+		Required:         true,
+		Description:      "The fully qualified name of the role which will be granted to the user or parent role.",
+		ForceNew:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+	},
+	"user_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Description:      "The fully qualified name of the user on which specified role will be granted.",
+		ForceNew:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		ExactlyOneOf: []string{
+			"user_name",
+			"parent_role_name",
+		},
+	},
+	"parent_role_name": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		Description:      "The fully qualified name of the parent role which will create a parent-child relationship between the roles.",
+		ForceNew:         true,
+		ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		ExactlyOneOf: []string{
+			"user_name",
+			"parent_role_name",
+		},
+	},
+}
+
+func GrantRole() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateGrantRole,
+		Read:   ReadGrantRole,
+		Delete: DeleteGrantRole,
+		Schema: grantRoleSchema,
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), helpers.IDDelimiter)
+				if len(parts) != 3 {
+					return nil, fmt.Errorf("invalid ID specified: %v, expected <role_name>|<grantee_object_type>|<grantee_identifier>", d.Id())
+				}
+				if err := d.Set("role_name", parts[0]); err != nil {
+					return nil, err
+				}
+				switch parts[1] {
+				case "ROLE":
+					if err := d.Set("parent_role_name", parts[2]); err != nil {
+						return nil, err
+					}
+				case "USER":
+					if err := d.Set("usere_name", parts[2]); err != nil {
```

Comment:
usere_name -> user_name

Reply:
It was not spotted by tests, so let's add an import step to both of the tests pls.